src/api_client/client.ts

```diff
@@ -209,135 +188,154 @@ export default class Client extends Api {
-    };
+    const body = this.buildJsonRpcRequest<CallRequest>(
+      JSONRPCMethod.METHOD_CALL,
+      { 
+        body: msg.body,
+        auth_type: msg.auth_type,
+        sender: msg.sender || '',
+      }
+    )
 
-    const res = await super.post<CallRes>(`/api/v1/call`, req);
+    const res = await super.post<JsonRPCResponse<CallResponse>>(`rpc/v1`, body)
 
     // if we get a 401, we need to return the response so we can try to authenticate
-    if (res.status !== 401) {
-      checkRes(res);
-    }
-
-    let result: any = null;
-
-    if (res.data.result) {
-      const uint8 = new Uint8Array(base64ToBytes(res.data.result));
-      const decoder = new TextDecoder('utf-8');
-      const jsonString = decoder.decode(uint8);
-      result = JSON.parse(jsonString);
+    if (res.status === 401) {
+      return {
+        status: res.status,
+        data: JSON.parse(bytesToString(base64ToBytes(res.data.result.result))),
+      }
     }
 
-    const cleanReceipt: MsgReceipt = !result
-      ? {
-          result: null,
-        }
-      : {
-          result: result,
-        };
+    return checkRes(res, (r) => {
+      return {
+        result: JSON.parse(bytesToString(base64ToBytes(r.result.result)))
+      }
+    });
+  }
 
+  private buildJsonRpcRequest<T>(method: JSONRPCMethod, params: T): JsonRPCRequest<T> {
     return {
-      status: res.status,
-      data: cleanReceipt,
+      jsonrpc: '2.0',
+      id: this.jsonRpcId++,
+      method,
+      params,
     };
   }
 }
 
 function checkRes<T, R>(
-  res: GenericResponse<T>,
-  selector?: (r: T) => R | undefined
+  res: AxiosResponse<JsonRPCResponse<T>>,
+  selector: (r: JsonRPCResponse<T>) => R | undefined
 ): GenericResponse<R> {
-  if (res.status != 200 || !res.data) {
-    throw new Error(
-      JSON.stringify(res.data) ||
+
+  switch (res.status) {
+    case 200:
+      break;
+    case 401:
+      throw new Error(JSON.stringify(res.data) || 'Unauthorized.');
+    case 404:
+      throw new Error(JSON.stringify(res.data) || 'Not found.');
+    case 500:
+      throw new Error(JSON.stringify(res.data) || 'Internal server error.');
+    default:
+      throw new Error(
+        JSON.stringify(res.data) ||
         'An unknown error has occurred.  Please check your network connection.'
-    );
+      );
+  }
+
+  if (!res.data) {
+    throw new Error(`failed to parse response: ${res}`);
+  }
+
+  if (res.data.error) {
+    const data = res.data.error.data ? `, data: ${JSON.stringify(res.data.error.data)}` : '';
+    throw new Error(`JSON RPC call error: code: ${res.data.error.code}, message: ${res.data.error.message}` + data);
+  }
+
+  if (res.data.jsonrpc !== '2.0') {
+    throw new Error(JSON.stringify(res.data) || 'Invalid JSON RPC response.');
   }
 
-  if (!selector) {
-    return { status: res.status, data: undefined };
+  if (!res.data.result) {
+    throw new Error(JSON.stringify(res.data) || 'No result in JSON RPC response.');
   }
 
   return {
```

Comment:
@jchappelow, do you mind taking a look at this `checkRes` function? I tried to model similar to how you showed me [here](https://github.com/jchappelow/kwil-db/blob/9528cd89e2702be1bbcebd848d60ceaa95e8edb4/core/rpc/client/user/jsonrpc/client.go#L134-L177).